program-test/tests/toolarge.rs

```diff
@@ -0,0 +1,31 @@
+use {
+    solana_program_test::ProgramTest,
+    solana_sdk::{
+        instruction::Instruction, packet::PACKET_DATA_SIZE, pubkey::Pubkey, signature::Signer,
+        transaction::Transaction,
+    },
+};
+
+#[tokio::test]
+async fn test_transaction_too_large() {
+    let (mut banks_client, payer, recent_blockhash) = ProgramTest::default().start().await;
+
+    let mut transaction = Transaction::new_with_payer(
+        &[Instruction {
+            program_id: Pubkey::default(),
+            accounts: vec![],
+            data: [42; PACKET_DATA_SIZE * 2].to_vec(),
+        }],
+        Some(&payer.pubkey()),
+    );
+    transaction.sign(&[&payer], recent_blockhash);
+
+    assert_eq!(
```

Comment:
I'm not proud of this assert.  Going from a `TransportError` back into a typed `TransactionError` seems hard, and I didn't feel like reworking the signature of `process_transaction` just for this

Reply:
Yeah, lack of PartialEq on TransportError is a pita; I've been running into that trying to update `tarpc` as well. I can't think of a better simple assert here.